### PR TITLE
Format DBNull values as NULL instead of empty string

### PR DESCRIPTION
**Pull request by alitas**
The formatter did not handle the value being of type `DBNull`, causing it to call `.ToString()` on the object, returning an empty string instead of `NULL`.

### CODE REVIEW

**Comment by shueybubbles** [comment]
thx for being our first public PR contributor!

Pls add a test that fails without this fix. Let me know if you need help putting together a `toolsconnectioninfo.xml` file with your connection string for the test to use.

Did you notice it with extended properties?  This function isn't used very many places.

**Comment by alitas** [comment]
> thx for being our first public PR contributor!
> 
> Pls add a test that fails without this fix. Let me know if you need help putting together a `toolsconnectioninfo.xml` file with your connection string for the test to use.
> 
> Did you notice it with extended properties? This function isn't used very many places.

We noticed it when scripting partition functions. I added a test case similar to what we encountered.

File: src/FunctionalTest/Smo/ScriptingTests/PartitionFunction_SmoTestSuite.cs
Comment (Matteo-T) [inline src/FunctionalTest/Smo/ScriptingTests/PartitionFunction_SmoTestSuite.cs:74]: Can't you just use an object initializer for those 2 properties

```csharp
new _SMO.PartitionFunction(db, "PF_NULL_RANGE_TEST") { RangeType = .. ; RangeValues = new object[] { ... } };
```

File: src/FunctionalTest/Smo/ScriptingTests/PartitionFunction_SmoTestSuite.cs
Comment (Matteo-T) [inline src/FunctionalTest/Smo/ScriptingTests/PartitionFunction_SmoTestSuite.cs:85]: Add a comment that inform whoever is running the test and getting an error about what went wrong.

Also, I don't know about you, but I always get confused by which param is "expected" and which one is "actual", just like you here :-)

Two options:
1. Assert.That(scripts.Length, Is.EqualTo(1), $"Oh no!");
2. Assert.AreEqual(expected: 1, actual: Length, "What?!?");

David and I like the fluent assertions a lot (he does a lot more and he is bad influence on me... :)

File: src/FunctionalTest/Smo/ScriptingTests/PartitionFunction_SmoTestSuite.cs
Comment (alitas) [reply src/FunctionalTest/Smo/ScriptingTests/PartitionFunction_SmoTestSuite.cs:74]: Sure that's possible. I tried to be consistent with the rest of the test suite.

**Review by Matteo-T — CHANGES_REQUESTED** [review]
Have a look at my comments and let me know what you think!

Again, thanks for being the 1st contributor! :)

File: src/FunctionalTest/Smo/ScriptingTests/PartitionFunction_SmoTestSuite.cs
Comment (shueybubbles) [inline src/FunctionalTest/Smo/ScriptingTests/PartitionFunction_SmoTestSuite.cs:88]:
```c# 
Assert.That(scripts[0], Contains.Substring("VALUES (NULL, 10, 100)"), "Incorrect script for null valued function");
```

We've adopted the nUnit constraints the past couple years because they provide very rich log output when they fail and are very expressive as written. The reason all the code doesn't have them is just that SMO is 20+ years old and we haven't tried to retrofit all of it to new styles. We would like all new code to follow the styles defined in .editorconfig and to use the NUnit constraints for asserts.
thx

**Review by shueybubbles — APPROVED** [review]
Will merge once the assert is changed. Thx a ton!
After it's merged we will include it in the next public SMO release that aligns with SSMS, in a couple months.

**Review by Matteo-T — APPROVED** [review]

**Comment by dzsquared** [comment]
Congrats @alitas on the first merged community PR to SMO!

**Comment by 0xSQL** [comment]
Thank you @alitas! And congrats!

**Pull request merged**
